src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -48,6 +48,80 @@ static class BitOperations
             19, 27, 23, 06, 26, 05, 04, 31
         };
 
+        /// <summary>
+        /// Evaluate whether a given integral value is a power of 2.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsPow2(int value)
+        {
+            if (Popcnt.IsSupported)
+            {
+                // 1 set bit means number is a power of 2 unless it is the sign bit, so get rid of that
+                // Cast to unsigned type so logic shift not signed arithmetic shift
+                return PopCount((uint)value << 1) == 1;
+            }
+
+            return SoftwareFallback((uint)value << 1);
+
+            static bool SoftwareFallback(uint value) => (value & (value - 1)) == 0 && value != 0;
```

Comment:
Given how small the software fallback is, this should likely just be inlined directly without a `SoftwareFallback` wrapper.

I'm hard pressed to think of a scenario where the ~5 instructions shouldn't just be inlined.

Reply:
I would keep it as is, so that the "SoftwareFallback-pattern" is obvious (and consistent).
At runtime the JIT will inline this tiny method.

Reply:
Good point. Would be good to validate it is actually being inlined, however.

Reply:
It is inlined (checked it just before commenting).